core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/FabricUtils.scala

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import spray.json._
+import scala.util.matching.Regex
+import scala.io.Source
+
+case class TokenServiceConfig(tokenServiceEndpoint: String,
+                              clusterType: String,
+                              clusterName: String,
+                              sessionToken: String)
+
+object TokenServiceConfigProtocol extends DefaultJsonProtocol {
+  implicit val TokenServiceConfigFormat: RootJsonFormat[TokenServiceConfig] = jsonFormat4(TokenServiceConfig)
+}
+
+import TokenServiceConfigProtocol._
+
+object FabricUtils {
+  private var TridentContext: Map[String, String] = Map[String, String]()
```

Comment:
dont use vars unless you have too, maybe you can make this a lazy val if you want to defer its used

Reply:
Needs to be var. Context is filled later in the file.

Reply:
i think it can be a lazy val at least, sometimes it takes a little cleverness to remove vars